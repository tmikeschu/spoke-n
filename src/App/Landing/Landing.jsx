import React from 'react'
import { Link } from 'react-router'
import contact from './contact_logo.png'
import podcast from './podcast_logo.png'
import map from './map_logo.png'
import instagram from './instagram_logo.png'

const Landing = () => (
  <section className="landing">
    <article className="logos">
      <Link to="map"><img src={map} alt="map"/></Link>
      <Link to="podcast"><img src={podcast} alt="podcast"/></Link>
      <Link to="contact"><img src={contact} alt="contact"/></Link>
      <Link to="instagram"><img src={instagram} alt="instagram"/></Link>
    </article>

    <article className="video">
      <a href="http://www.5280.com/2017/05/du-alumni-to-bike-from-denver-to-patagonia/" target="_blank" rel="noopener noreferrer">
        Recently featured in Denver's 5280!
      </a>
      <iframe title="Episode 1" src="https://player.vimeo.com/video/211431209?byline=0&portrait=0"
        width="640" height="360" frameBorder="0"
        webkitallowfullscreen mozallowfullscreen allowFullScreen>
      </iframe>
      <iframe title="Episode 2" src="https://player.vimeo.com/video/207747360?byline=0&portrait=0"
        width="640" height="360" frameBorder="0"
        webkitallowfullscreen mozallowfullscreen allowFullScreen>
      </iframe>
      <iframe title="Episode 3" src="https://player.vimeo.com/video/211432545?byline=0&portrait=0"
        width="640" height="360" frameBorder="0"
        webkitallowfullscreen mozallowfullscreen allowfullscreen>
      </iframe>
      <iframe title="Episode 4" src="https://player.vimeo.com/video/215738304?byline=0&portrait=0"
        width="640" height="360" frameBorder="0"
        webkitallowfullscreen mozallowfullscreen allowFullScreen>
      </iframe>
    </article>
  </section>
)

export default Landing

