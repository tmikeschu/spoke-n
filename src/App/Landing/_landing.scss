.Landing {
  &__wrapper {
    @include contentContainer;
    background-color: transparent;
    box-shadow: none;
  }
  .logos {
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin: 0 auto;
    width: 100%;
    a {
      background-color: rgba(256, 256, 256, 0.7);
      @include boxShadow;
      max-width: 200px;
      margin: 0 8px;
      &:hover,
      &:focus {
        border: none;
        img {
          transition: transform 500ms;
          transform: scale(1.1);
        }
      }

      img {
        width: 100%;
        border-radius: 5px;
        transition: transform 500ms;
      }
    }
  }
  .video {
    display: flex;
    flex-direction: column;
    align-items: center;
    a {
      margin: 16px 0;
      border-radius: 5px;
      background-color: rgba(256, 256, 256, 0.7);
      @include boxShadow;
      font-size: 40px;
      line-height: 1.5em;
      padding: 8px;
      box-sizing: border-box;
    }
    iframe {
      margin: 16px 0;
    }
  }

  @media (max-width: 640px) {
    .logos {
      width: 100%;
      max-width: 416px;
      flex-wrap: wrap;
      justify-items: center;
      a {
        width: 48%;
        margin: 0 1%;
        margin-bottom: 10px;
        border-radius: 3px;
        img {
          width: 100%;
        }
      }
    }

    .video {
      iframe {
        margin: 0;
        width: 100%;
      }
    }
  }
}
