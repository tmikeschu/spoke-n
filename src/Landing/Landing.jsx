import React, { Component } from 'react';
import { Link } from 'react-router';
import contact from '../contact_logo.jpg';
import podcast from '../podcast_logo.jpg';
import map from '../map_logo.jpg';
import instagram from '../instagram_logo.jpg';
import '../stylesheets/App.css';

export default class Landing extends Component {
  render() {
    return (
      <section className="landing">
        <article>
          <Link to="map"><img src={map} alt="map"/></Link>
          <Link to="podcast"><img src={podcast} alt="podcast"/></Link>
          <Link to="contact"><img src={contact} alt="contact"/></Link>
          <Link to="instagram"><img src={instagram} alt="instagram"/></Link>
        </article>
      </section>
    )
  }
}
