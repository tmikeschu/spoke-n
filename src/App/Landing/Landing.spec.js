import React from "react"
import { shallow } from "enzyme"
import Landing from "./Landing"

describe("<Landing />", () => {
  const landing = shallow(<Landing />)

  it("renders without crashing", () => {
    expect(landing).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(landing.getNodes()).toMatchSnapshot()
    })
  })
})

