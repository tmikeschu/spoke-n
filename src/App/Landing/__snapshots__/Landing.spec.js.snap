// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> snapshot is valid 1`] = `
Array [
  <section
    className="landing"
>
    <article
        className="logos"
    >
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="map"
        >
            <img
                alt="map"
                src="map_logo.png"
            />
        </Link>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="podcast"
        >
            <img
                alt="podcast"
                src="podcast_logo.png"
            />
        </Link>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="contact"
        >
            <img
                alt="contact"
                src="contact_logo.png"
            />
        </Link>
        <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="instagram"
        >
            <img
                alt="instagram"
                src="instagram_logo.png"
            />
        </Link>
    </article>
    <article
        className="video"
    >
        <iframe
            allowFullScreen={true}
            frameBorder="0"
            height="360"
            mozallowfullscreen={true}
            src="https://player.vimeo.com/video/211431209?byline=0&portrait=0"
            title="Episode 1"
            webkitallowfullscreen={true}
            width="640"
        />
        <iframe
            allowFullScreen={true}
            frameBorder="0"
            height="360"
            mozallowfullscreen={true}
            src="https://player.vimeo.com/video/207747360?byline=0&portrait=0"
            title="Episode 2"
            webkitallowfullscreen={true}
            width="640"
        />
        <iframe
            allowfullscreen={true}
            frameBorder="0"
            height="360"
            mozallowfullscreen={true}
            src="https://player.vimeo.com/video/211432545?byline=0&portrait=0"
            title="Episode 3"
            webkitallowfullscreen={true}
            width="640"
        />
    </article>
</section>,
]
`;
